internal/client/client_batch.go

```diff
@@ -549,14 +549,15 @@ func (c *batchCommandsClient) waitConnReady() (err error) {
 		metrics.TiKVBatchClientWaitEstablish.Observe(time.Since(start).Seconds())
 	}()
 	dialCtx, cancel := context.WithTimeout(context.Background(), c.dialTimeout)
+	// Trigger idle connection to reconnection
+	// Put it outside loop to avoid unnecessary reconnecting.
+	c.conn.Connect()
```

Comment:
How about to check the state here? We only need to call `Connect` when the state is IDLE.

Reply:
Ok, done.